cmd/analyzer/analyzer.go

```diff
@@ -173,52 +232,88 @@ func NewService(cfg *config.AnalysisConfig) (*Service, error) {
 	analyzers := map[string]A{}
 	if cfg.Analyzers.Consensus != nil {
 		analyzers, err = addAnalyzer(analyzers, err, func() (A, error) {
-			return consensus.NewMain(&cfg.Source, cfg.Analyzers.Consensus, client, logger)
+			chainContext := cfg.Source.History().CurrentRecord().ChainContext
+			sourceClient, err1 := sources.Consensus(ctx)
+			if err1 != nil {
+				return nil, err1
+			}
+			return consensus.NewMain(cfg.Analyzers.Consensus, chainContext, sourceClient, dbClient, logger)
 		})
 	}
 	if cfg.Analyzers.Emerald != nil {
 		analyzers, err = addAnalyzer(analyzers, err, func() (A, error) {
-			return runtime.NewRuntimeAnalyzer(common.RuntimeEmerald, &cfg.Source, cfg.Analyzers.Emerald, client, logger)
+			sdkPT := cfg.Source.SDKParaTime(common.RuntimeEmerald)
+			sourceClient, err1 := sources.Runtime(ctx, common.RuntimeEmerald)
+			if err1 != nil {
+				return nil, err1
+			}
```

Comment:
I'm a little unhappy about the code duplication here, but I don't have great suggestions.

Refloating a recent idea: If `RuntimeClient` used lazy initialization, we could inline `sources.Runtime` and not worry about errors. It would also make it so the `query_on_cache_miss` flag is not needed.
Definitely not something we'd do in this PR, and likely ever, just from the effort/benefits ratio. But a man can dream.